modules/ROOT/pages/dataweave-create-module.adoc

```diff
@@ -181,3 +181,278 @@ WeaveMod::myFunc("dataweave")
+*
+* The input description (optional)
+*
+* [source,%The language%,linenums] (optional)
+* ----
+* YOUR CODE
+* ----
+*
+* ==== Output (optional section)
+*
+* %The output description% (optional)
+*
+* [source,%The language%,linenums] (optional)
+* ----
+* YOUR CODE
+* ----
+*/
+----
+
+==== Description Section
+
+This section provides a description of the function. The section starts at the top of the comments and extends to the line before the first section heading. If there is no section header, the description extends to the end of the comments for the function.
+
+A description consists of the following parts:
+
+* Short description: The first paragraph, which provides the primary description of the function.
+* Long description: Any additional information about the function. This optional description begins two line breaks below the short description.
+
+See the <<example-documentation, example>> that includes a description with sections for the parameters and examples.
+
+Each part is useful for auto-generated documentation.
+
+==== Parameters Section
+
+The optional `=== Parameters` section describes parameters of the function.
+
+The template uses Asciidoc table format to document each parameter in a table row.
+
+....
+[%header, cols="1,1,3"]
+|===
+| Name | Type | Description
+| yourParameter1 | The parameter type | Your description here.
+| yourParameter2 | The parameter type | Your description here.
+|===
+....
+
+==== Example Section
+
+You can use zero or more `=== Example` sections to provide any examples needed to illustrate how your function works.
+
+This section contains the following optional subsections:
+
+* `==== Input` for input to the DataWeave script.
+* `==== Source` for the DataWeave script.
+* `==== Output` for output generated by the script.
+
+All subsections follow the same template, which includes an optional description and an optional code section:
+
+....
+Your section description
+
+[source,%The language%,linenums]
+----
+YOUR CODE
+----
+....
+
+
+[[example-documentation]]
+The following example shows the xref:dw-core-functions-log.adoc[log] function documentation.
+
+....
+/**
+* Without changing the value of the input, `log` returns the input as a system
+* log. So this makes it very simple to debug your code, because any expression or subexpression can be wrapped
+* with *log* and the result will be printed out without modifying the result of the expression.
+* The output is going to be printed in application/dw format.
+* (1)
+*
+* The prefix parameter is optional and allows to easily find the log output.
+*
+*
+* Use this function to help with debugging DataWeave scripts. A Mule app
+* outputs the results through the `DefaultLoggingService`, which you can see
+* in the Studio console.
+*
+* === Parameters
+*
+* [%header, cols="1,13"]
+* |===
+* | Name | Type | Description
+* | `prefix` | `String` | An optional string that typically describes the log.
+* | `value` | `TODO` | The value to log.
```

Comment:
@mlischetti what type should we put here?

Reply:
In this case, T is the parameter. Because it's a parameterized type